EndlessClient/Subscribers/GuildEventSubscriber.cs

```diff
@@ -0,0 +1,54 @@
+﻿using AutomaticTypeMapper;
+using EndlessClient.Audio;
+using EndlessClient.Dialogs.Factories;
+using EOLib.Domain.Interact.Guild;
+using EOLib.Domain.Notifiers;
+using EOLib.Localization;
+using EOLib.Net.Communication;
+using Moffat.EndlessOnline.SDK.Protocol.Net.Client;
+
+namespace EndlessClient.Subscribers
+{
+    [AutoMappedType]
+    public class GuildEventSubscriber : IGuildNotifier
+    {
+        private readonly IEOMessageBoxFactory _messageBoxFactory;
+        private readonly ILocalizedStringFinder _localizedStringFinder;
+        private readonly IPacketSendService _packetSendService;
+        private readonly ISfxPlayer _sfxPlayer;
+
+        public GuildEventSubscriber(IEOMessageBoxFactory messageBoxFactory, ILocalizedStringFinder localizedStringFinder, IPacketSendService packetSendService, ISfxPlayer sfxPlayer)
+        {
+            _messageBoxFactory = messageBoxFactory;
+            _localizedStringFinder = localizedStringFinder;
+            _packetSendService = packetSendService;
+            _sfxPlayer = sfxPlayer;
+        }
+
+        public void NotifyGuildCreationRequest(GuildCreationRequest request)
+        {
+            _sfxPlayer.PlaySfx(SoundEffectID.ServerMessage);
+
+            var dlg = _messageBoxFactory.CreateMessageBox(
+                $"{request.GuildIdentity}" + " " +
+                _localizedStringFinder.GetString(DialogResourceID.GUILD_INVITES_YOU_TO_JOIN) + " " +
+                _localizedStringFinder.GetString(EOResourceID.GUILD_JOINING_A_GUILD_IS_FREE) + " " +
+                _localizedStringFinder.GetString(EOResourceID.GUILD_PLEASE_CONSIDER_CAREFULLY) + " " +
+                _localizedStringFinder.GetString(EOResourceID.GUILD_DO_YOU_ACCEPT)
+                , caption: _localizedStringFinder.GetString(DialogResourceID.GUILD_INVITATION), whichButtons: Dialogs.EODialogButtons.OkCancel,
```

Comment:
I'm really torn on this. Every other `DialogResourceID` specifies the first resource and uses both as a pair. This is kind of a special case because there's so much extra text.

If you take a look at the overloads for `EOMessageBoxFactory.CreateMessageBox` you'll see that there are some that allow putting extra text in there.

If you wouldn't mind, I'd appreciate if you could investigate how ugly it would be to use/add one that incorporates the additional text that this dialog box requires. If you're spending more than like 10 minutes on it I'm fine with accepting this as it is, but I'd like to avoid introducing the second resource for the pair of `DialogResourceID` values if at all possible.

Reply:
Yeah the one that accepts extra data isn't really going to work without modification since it prepends the guild identity before the second dialog resource string :\

Reply:
Thanks for looking into it - that's kind of what I figured